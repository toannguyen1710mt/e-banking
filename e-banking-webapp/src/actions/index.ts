export * from './auth';
export * from './transaction';
export * from './account';
