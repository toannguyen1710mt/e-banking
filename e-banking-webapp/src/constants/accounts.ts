// Interfaces
import { AccountType } from '@/interfaces';

export const ACCOUNT_TYPES = {
  MAIN_WALLET: 'Main Wallet',
  SAVINGS: 'Savings',
  CHECKING: 'Checking',
};

export const TRANSFER_FORM_ACCOUNT_OPTIONS = [
  {
    key: AccountType.MAIN,
    label: 'Main Account',
  },
  {
    key: AccountType.SAVINGS,
    label: 'Savings',
  },
  {
    key: AccountType.CHECKING,
    label: 'Checking',
  },
];
