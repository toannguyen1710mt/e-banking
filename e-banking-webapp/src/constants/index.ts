export * from './accounts';
export * from './routes';
export * from './themes';
export * from './messages';
export * from './images';
export * from './regex';
export * from './navbar';
export * from './titles';
export * from './global-transfer';
export * from './balance';
export * from './charts';
export * from './form';
export * from './api';
export * from './tags';
export * from './settings';
export * from './tables';
export * from './cards';
export * from './date';
export * from './sidebar-list';
