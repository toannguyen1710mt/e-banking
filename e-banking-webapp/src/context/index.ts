export * from './WizardContext';
export * from './ToastContext';
export * from './FetchedBalancesContext';
export * from './useUserContext';
