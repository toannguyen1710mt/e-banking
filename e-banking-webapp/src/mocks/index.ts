export * from './options';
export * from './card';
export * from './charts';
export * from './transaction';
export * from './table';
export * from './transferConfirmData';
export * from './target';
export * from './analytics';
export * from './auth';
export * from './ServiceCard';
export * from './formMocks';
