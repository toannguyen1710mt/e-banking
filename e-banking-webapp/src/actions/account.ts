'use server';

// Libs
import { revalidateTag } from 'next/cache';

// Constants
import { API_ENDPOINTS, ERROR_MESSAGES } from '@/constants';

// Interfaces
import { IAccountPayloadData } from '@/interfaces';

// Services
import { httpClient } from '@/services';

export const updateAccountInfo = async (
  accountDocumentId: string,
  accountData: IAccountPayloadData,
) => {
  try {
    await httpClient.put(`${API_ENDPOINTS.ACCOUNTS}/${accountDocumentId}`, {
      data: accountData,
    });

    revalidateTag(API_ENDPOINTS.ACCOUNTS);
  } catch (error) {
    throw new Error(ERROR_MESSAGES.ERROR_UPDATE_ACCOUNT_INFO + error);
  }
};
