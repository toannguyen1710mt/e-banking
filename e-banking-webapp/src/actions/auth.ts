'use server';

import {
  signIn,
  signOut as nextAuthSignOut,
  unstable_update,
} from '@/config/auth';
import { revalidateTag } from 'next/cache';
import { AuthError } from 'next-auth';

// Constants
import { ERROR_MESSAGES, API_ENDPOINTS, TAGS } from '@/constants';

// Services
import { httpClient } from '@/services/http-client';

// Interfaces
import {
  TSignUpPayload,
  TSignInFormData,
  TUpdateInfo,
  AuthResponse,
  IAccount,
  IAccountPayload,
  ICardPayload,
  SuccessResponse,
  TChangePasswordFormData,
  TChangePasswordSuccessResponse,
  IUser,
  ICard,
} from '@/interfaces';

export const authenticateUser = async (formData: TSignInFormData) => {
  try {
    await signIn('credentials', formData);
  } catch (error) {
    if (error instanceof AuthError) {
      return ERROR_MESSAGES.ACCOUNT_AND_PASSWORD_INVALID;
    }
    throw error;
  }
};

export const signUp = async (signUpData: TSignUpPayload) => {
  const response = await httpClient.post<AuthResponse>(
    API_ENDPOINTS.SIGN_UP,
    signUpData,
  );

  return response;
};

export const signOut = async () => await nextAuthSignOut({ redirect: false });

export const changePassword = async (
  payload: TChangePasswordFormData,
  jwt: string,
) => {
  const response = await httpClient.post<TChangePasswordSuccessResponse>(
    API_ENDPOINTS.CHANGE_PASSWORD,
    payload,
    {
      headers: {
        Authorization: `Bearer ${jwt}`,
      },
    },
  );

  return response;
};

export const updateUser = async (id: number, payload: TUpdateInfo) => {
  try {
    const response = await httpClient.put(
      `${API_ENDPOINTS.USERS}/${id}`,
      payload,
    );

    await unstable_update({ user: response.data });

    return response.data;
  } catch (error) {
    if (error instanceof AuthError) {
      throw ERROR_MESSAGES.SIGN_UP_ERROR;
    }
  }
};

export const addAccount = async (payload: IAccountPayload) => {
  try {
    const response = await httpClient.post<SuccessResponse<IAccount>>(
      API_ENDPOINTS.ACCOUNTS,
      payload,
    );

    return response.data;
  } catch (error) {
    if (error instanceof AuthError) {
      throw ERROR_MESSAGES.SIGN_UP_ERROR;
    }
  }
};

export const addCard = async (payload: ICardPayload) => {
  try {
    const response = await httpClient.post(API_ENDPOINTS.CARDS, payload);

    return response.data;
  } catch (error) {
    if (error instanceof AuthError) {
      throw ERROR_MESSAGES.SIGN_UP_ERROR;
    }
  }
};

export const updateEmailSettings = async (
  userId: number,
  payload: Partial<IUser>,
) => {
  await httpClient.put(`${API_ENDPOINTS.USERS}/${userId}`, payload);

  return revalidateTag(TAGS.USERS);
};

export const getUser = async (id: number): Promise<IUser | undefined> => {
  try {
    const response = await httpClient.get(`${API_ENDPOINTS.USERS}/${id}`);

    return response.data;
  } catch (error) {
    if (error instanceof AuthError) {
      throw ERROR_MESSAGES.SIGN_UP_ERROR;
    }
  }
};

export const addNewCardByAccountId = async (
  accountId: string,
  payload: Omit<ICard, 'id' | 'fullName' | 'account'>,
) => {
  try {
    const response = await httpClient.post(API_ENDPOINTS.CARDS, {
      data: {
        ...payload,
        account: {
          connect: [accountId],
        },
      },
    });

    revalidateTag(TAGS.CARD);

    return response.data;
  } catch (error) {
    if (error instanceof AuthError) {
      throw ERROR_MESSAGES.ADD_CARD_FAILED;
    }
  }
};
