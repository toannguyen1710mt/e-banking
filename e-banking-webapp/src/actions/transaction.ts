'use server';

// Libs
import { revalidateTag } from 'next/cache';

// Constants
import { API_ENDPOINTS, ERROR_MESSAGES } from '@/constants';

// Interfaces
import { ITransaction, TransactionCreateData } from '@/interfaces';

// Services
import { httpClient } from '@/services';

export const createTransaction = async (
  accountId: string,
  transactionData: TransactionCreateData,
) => {
  try {
    await httpClient.post<ITransaction>(API_ENDPOINTS.TRANSACTIONS, {
      data: {
        ...transactionData,
        account: {
          connect: [accountId],
        },
      },
    });

    revalidateTag(API_ENDPOINTS.TRANSACTIONS);
  } catch (error) {
    throw new Error(ERROR_MESSAGES.ERROR_CREATE_TRANSACTION + error);
  }
};
