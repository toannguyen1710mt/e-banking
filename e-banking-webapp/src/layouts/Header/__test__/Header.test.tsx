import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { redirect, usePathname } from 'next/navigation';

// Context
import { ToastProvider, UserProvider } from '@/context';

// Mocks
import { MOCK_SESSION_DATA } from '@/mocks';

// Actions
import { signOut } from '@/actions';

// Constants
import { ERROR_MESSAGES } from '@/constants';

// Components
import { Header } from '..';

jest.mock('@/actions', () => ({
  signOut: jest.fn(),
}));

jest.mock('next/navigation', () => ({
  redirect: jest.fn(),
  usePathname: jest.fn(() => '/sign-in'),
}));

const mockShowToast = jest.fn();

jest.mock('@/context', () => ({
  ...jest.requireActual('@/context'),
  ToastProvider: ({ children }: { children: React.ReactNode }) => {
    return <div>{children}</div>;
  },
  useToastContext: () => ({
    showToast: mockShowToast,
  }),
}));

describe('Header component', () => {
  const renderWithProviders = (ui: React.ReactNode) => {
    return render(
      <UserProvider avatar={''}>
        <ToastProvider>{ui}</ToastProvider>
      </UserProvider>,
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();
    (usePathname as jest.Mock).mockReturnValue('/');
  });

  test('matches the snapshot', () => {
    const { container } = renderWithProviders(
      <Header session={MOCK_SESSION_DATA} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('toggles the mobile menu on click', () => {
    renderWithProviders(<Header session={MOCK_SESSION_DATA} />);
    // Find menu toggle button
    const menuToggle = screen.getByLabelText('Open menu');
    // Click to open menu
    fireEvent.click(menuToggle);

    expect(menuToggle).toHaveAttribute('aria-label', 'Close menu');

    // Click again to close menu
    fireEvent.click(menuToggle);
    expect(menuToggle).toHaveAttribute('aria-label', 'Open menu');
  });

  test('should call signOut and redirect on successful sign out', async () => {
    (signOut as jest.Mock).mockResolvedValueOnce(undefined);

    renderWithProviders(<Header session={MOCK_SESSION_DATA} />);

    // Step 1: Open the dropdown menu by clicking the avatar
    const avatarButton = screen.getByRole('img', {
      name: /avatar/i,
    });
    fireEvent.click(avatarButton);

    // Step 2: Query for the "Sign out" button inside the dropdown
    const signOutButton = await screen.findByRole('button', {
      name: /sign out/i,
    });

    // Step 3: Click the "Sign out" button
    fireEvent.click(signOutButton);

    // Step 4: Verify loading state and behavior
    await waitFor(() => expect(signOutButton).toBeDisabled());

    await waitFor(() => {
      expect(signOut).toHaveBeenCalled();
      expect(mockShowToast).toHaveBeenCalledWith(
        ERROR_MESSAGES.SIGN_OUT_SUCCESS,
        'success',
        'top-center',
      );
      expect(redirect).toHaveBeenCalledWith('/sign-in');
    });
  });

  test('should handle sign out failure', async () => {
    (signOut as jest.Mock).mockRejectedValueOnce(new Error('Sign out failed'));

    renderWithProviders(<Header session={MOCK_SESSION_DATA} />);

    // Step 1: Open the dropdown menu by clicking the avatar
    const avatarButton = screen.getByRole('img', {
      name: /avatar/i,
    });
    fireEvent.click(avatarButton);

    // Step 2: Query for the "Sign out" button inside the dropdown
    const signOutButton = await screen.findByRole('button', {
      name: /sign out/i,
    });

    fireEvent.click(signOutButton);

    await waitFor(() => {
      expect(signOut).toHaveBeenCalled();
      expect(mockShowToast).toHaveBeenCalledWith(
        ERROR_MESSAGES.SIGN_OUT_FAILED,
        'error',
        'top-center',
      );
    });
  });
});
