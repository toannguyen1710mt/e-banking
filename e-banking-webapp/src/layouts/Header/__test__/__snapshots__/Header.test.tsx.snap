// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot 1`] = `
<div>
  <div>
    <nav
      class="flex z-40 w-full h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bg-background/70"
      style="--navbar-height: 4rem;"
    >
      <header
        class="z-40 flex gap-4 w-full flex-row relative flex-nowrap items-center h-[var(--navbar-height)] max-w-full justify-start border-b border-foreground-100 border-opacity-25 px-3 bg-background-900"
      >
        <div
          class="flex items-center gap-[88px]"
        >
          <div
            class="flex basis-0 flex-row flex-grow flex-nowrap justify-start bg-transparent items-center no-underline text-medium whitespace-nowrap box-border gap-5"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Open menu"
              aria-pressed="false"
              class="group items-center justify-center w-6 h-full tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased text-primary-200 rounded-lg bg-background-500 border-1 data-[focus=true]:border-primary data-[focus=true]:border-1 cursor-pointer hidden max-[700px]:flex"
              data-slot="trigger"
              id="react-aria-:r0:"
              type="button"
            >
              <span
                class="sr-only"
              >
                open navigation menu
              </span>
              <span
                class="w-full h-full pointer-events-none flex flex-col items-center justify-center text-inherit group-data-[pressed=true]:opacity-70 transition-opacity before:content-[''] before:block before:h-px before:w-6 before:bg-current before:transition-transform before:duration-150 before:-translate-y-1 before:rotate-0 group-data-[open=true]:before:translate-y-px group-data-[open=true]:before:rotate-45 after:content-[''] after:block after:h-px after:w-6 after:bg-current after:transition-transform after:duration-150 after:translate-y-1 after:rotate-0 group-data-[open=true]:after:translate-y-0 group-data-[open=true]:after:-rotate-45"
              />
            </button>
            <a
              href="/"
            >
              <img
                alt="Logo EBanking"
                data-nimg="1"
                decoding="async"
                height="36"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
                style="color: transparent;"
                width="104"
              />
            </a>
          </div>
          <ul
            class="h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 flex gap-[59px] font-normal text-transparentBlack max-[700px]:hidden"
            data-justify="center"
          >
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
            >
              <a
                class="font-bold text-black"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
            >
              <a
                class="text-gray-500"
                href="/analytics"
              >
                Analytics
              </a>
            </li>
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold"
            >
              <a
                class="text-gray-500"
                href="/transactions"
              >
                Transactions
              </a>
            </li>
          </ul>
        </div>
        <ul
          class="flex gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0"
          data-justify="end"
        >
          <button
            aria-label="search icon"
            class="cursor-not-allowed"
            disabled=""
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                filter="url(#filter0_i_1_958)"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="11.4"
                  stroke="#F5F5F5"
                  stroke-width="1.2"
                />
              </g>
              <path
                d="M16.8552 16.145L15.0002 14.305C15.7202 13.4072 16.0689 12.2676 15.9746 11.1206C15.8803 9.9736 15.35 8.90634 14.493 8.13827C13.6359 7.37021 12.5171 6.95971 11.3667 6.99119C10.2162 7.02267 9.12156 7.49373 8.30777 8.30752C7.49398 9.12131 7.02291 10.216 6.99143 11.3664C6.95995 12.5169 7.37045 13.6356 8.13852 14.4927C8.90659 15.3498 9.97385 15.88 11.1208 15.9744C12.2678 16.0687 13.4074 15.72 14.3052 15L16.1452 16.84C16.1917 16.8868 16.247 16.924 16.3079 16.9494C16.3688 16.9748 16.4342 16.9879 16.5002 16.9879C16.5662 16.9879 16.6316 16.9748 16.6925 16.9494C16.7534 16.924 16.8087 16.8868 16.8552 16.84C16.9453 16.7467 16.9957 16.6221 16.9957 16.4925C16.9957 16.3628 16.9453 16.2382 16.8552 16.145ZM11.5002 15C10.808 15 10.1313 14.7947 9.5557 14.4101C8.98013 14.0255 8.53152 13.4789 8.26662 12.8393C8.00171 12.1998 7.9324 11.4961 8.06745 10.8171C8.2025 10.1382 8.53584 9.51456 9.02532 9.02508C9.51481 8.53559 10.1384 8.20225 10.8174 8.0672C11.4963 7.93216 12.2 8.00147 12.8396 8.26637C13.4791 8.53128 14.0258 8.97988 14.4103 9.55546C14.7949 10.131 15.0002 10.8077 15.0002 11.5C15.0002 12.4282 14.6314 13.3184 13.9751 13.9748C13.3187 14.6312 12.4285 15 11.5002 15Z"
                fill="#808080"
              />
              <defs>
                <filter
                  color-interpolation-filters="sRGB"
                  filterUnits="userSpaceOnUse"
                  height="24"
                  id="filter0_i_1_958"
                  width="24"
                  x="0"
                  y="0"
                >
                  <feflood
                    flood-opacity="0"
                    result="BackgroundImageFix"
                  />
                  <feblend
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    mode="normal"
                    result="shape"
                  />
                  <fecolormatrix
                    in="SourceAlpha"
                    result="hardAlpha"
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                  />
                  <feoffset />
                  <fegaussianblur
                    stdDeviation="1"
                  />
                  <fecomposite
                    in2="hardAlpha"
                    k2="-1"
                    k3="1"
                    operator="arithmetic"
                  />
                  <fecolormatrix
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.18 0"
                  />
                  <feblend
                    in2="shape"
                    mode="normal"
                    result="effect1_innerShadow_1_958"
                  />
                </filter>
              </defs>
            </svg>
          </button>
          <button
            aria-label="notification icon"
            class="cursor-not-allowed"
            disabled=""
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                filter="url(#filter0_i_1_1143)"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="11.4"
                  stroke="#F5F5F5"
                  stroke-width="1.2"
                />
              </g>
              <path
                d="M15 10C15 9.20435 14.6839 8.44129 14.1213 7.87868C13.5587 7.31607 12.7956 7 12 7C11.2044 7 10.4413 7.31607 9.87868 7.87868C9.31607 8.44129 9 9.20435 9 10C9 13.5 7.5 14.5 7.5 14.5H16.5C16.5 14.5 15 13.5 15 10Z"
                fill="#808080"
              />
              <path
                d="M12.8648 16.5C12.7769 16.6515 12.6507 16.7773 12.4989 16.8648C12.3471 16.9522 12.175 16.9982 11.9998 16.9982C11.8246 16.9982 11.6525 16.9522 11.5007 16.8648C11.3488 16.7773 11.2227 16.6515 11.1348 16.5"
                fill="#808080"
              />
              <defs>
                <filter
                  color-interpolation-filters="sRGB"
                  filterUnits="userSpaceOnUse"
                  height="24"
                  id="filter0_i_1_1143"
                  width="24"
                  x="0"
                  y="0"
                >
                  <feflood
                    flood-opacity="0"
                    result="BackgroundImageFix"
                  />
                  <feblend
                    in="SourceGraphic"
                    in2="BackgroundImageFix"
                    mode="normal"
                    result="shape"
                  />
                  <fecolormatrix
                    in="SourceAlpha"
                    result="hardAlpha"
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                  />
                  <feoffset />
                  <fegaussianblur
                    stdDeviation="1"
                  />
                  <fecomposite
                    in2="hardAlpha"
                    k2="-1"
                    k3="1"
                    operator="arithmetic"
                  />
                  <fecolormatrix
                    type="matrix"
                    values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.18 0"
                  />
                  <feblend
                    in2="shape"
                    mode="normal"
                    result="effect1_innerShadow_1_1143"
                  />
                </filter>
              </defs>
            </svg>
          </button>
          <span
            aria-expanded="false"
            aria-haspopup="true"
            class="flex relative justify-center items-center box-border overflow-hidden align-middle outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased text-primary-200 bg-background-500 border-1 data-[focus=true]:border-primary data-[focus=true]:border-1 cursor-pointer h-6 w-6 rounded-full"
            data-slot="trigger"
            id="react-aria-:r3:"
            type="button"
          >
            <span
              aria-label="avatar"
              class="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex items-center justify-center text-inherit w-full h-full"
              role="img"
            >
              <svg
                aria-hidden="true"
                fill="none"
                height="80%"
                role="presentation"
                viewBox="0 0 24 24"
                width="80%"
              >
                <path
                  d="M12 2C9.38 2 7.25 4.13 7.25 6.75C7.25 9.32 9.26 11.4 11.88 11.49C11.96 11.48 12.04 11.48 12.1 11.49C12.12 11.49 12.13 11.49 12.15 11.49C12.16 11.49 12.16 11.49 12.17 11.49C14.73 11.4 16.74 9.32 16.75 6.75C16.75 4.13 14.62 2 12 2Z"
                  fill="currentColor"
                />
                <path
                  d="M17.0809 14.1489C14.2909 12.2889 9.74094 12.2889 6.93094 14.1489C5.66094 14.9989 4.96094 16.1489 4.96094 17.3789C4.96094 18.6089 5.66094 19.7489 6.92094 20.5889C8.32094 21.5289 10.1609 21.9989 12.0009 21.9989C13.8409 21.9989 15.6809 21.5289 17.0809 20.5889C18.3409 19.7389 19.0409 18.5989 19.0409 17.3589C19.0309 16.1289 18.3409 14.9889 17.0809 14.1489Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </span>
        </ul>
      </header>
    </nav>
  </div>
</div>
`;
